rskj-core/src/main/java/co/rsk/validators/BlockTxsValidationRule.java

```diff
@@ -76,21 +76,20 @@ public boolean isValid(Block block, Block parent) {
                 return false;
             }
 
-            byte[] txSender = tx.getSender();
-            ByteArrayWrapper key = new ByteArrayWrapper(txSender);
-            BigInteger expectedNonce = curNonce.get(key);
+            RskAddress sender = tx.getSender();
+            BigInteger expectedNonce = curNonce.get(sender);
             if (expectedNonce == null) {
-                expectedNonce = parentRepo.getNonce(txSender);
+                expectedNonce = parentRepo.getNonce(sender.getBytes());
```

Comment:
Maybe `org.ethereum.core.Repository` could be refactored as part of this work, it has a lot of `byte[]` referring RskAddresses

Reply:
I noticed that too, but the number of changes was getting out of control. Just created #410 to track that.